openhtf/__init__.py

```diff
@@ -394,6 +394,28 @@ def with_args(self, **kwargs):
     new_info.measurements = [m.with_args(**kwargs) for m in self.measurements]
     return new_info
 
+  def WithPlugs(self, **subplugs):
+    """Substitute plugs for placeholders for this phase."""
+    # Make a copy so we can have multiple of the same phase with different plugs
+    # in the same test.
+    new_info = mutablerecords.CopyRecord(self)
+    
+    for plug_name, plug_class in subplugs.items():
+      placeholders = [(i, p) for i, p in enumerate(self.plugs) 
```

Comment:
there can only be one plug with this name, right? This may benefit from taking advantage of that:

```py
for subplug_name, subplug_class in subplugs.items():
  for i, placeholder_plug in enumerate(self.plugs):
    if placeholder_plug.name == subplug_name:
      if not isinstance(placeholder_plug, plugs.BasePlug.Placeholder)):
        raise plugs.InvalidPlugError('Plug named %s was not a placeholder for %s' % (
            subplug_name, subplug_class))
      new_info.plugs[i] = mutablerecords.CopyRecord(placeholder_plug, cls=subplug_class)
      break
  else:
    raise plugs.InvalidPlugError('No plug named %s required for phase %s' % (
        subplug_name, self.name))
```

That might be worse. It definitely needs comments